src/parachain/redeem.ts

```diff
@@ -199,7 +193,7 @@ export class DefaultRedeemAPI {
         if (!this.isRequestSuccessful(result.events)) {
             throw new Error("Request failed");
         }
-        const id = this.getRedeemIdFromEvents(result.events, "RequestRedeem");
+        const id = this.getRedeemIdFromEvents(result.events, this.api.events.redeem.RequestRedeem);
         return { id, vault };
```

Comment:
Here as well, please return the redeem request. From this, we have all the info we need in the UI. I think it's also the logical thing to return from the request call.